isobus/src/isobus_virtual_terminal_client.cpp

```diff
@@ -4288,20 +4325,31 @@ namespace isobus
 		return retVal;
 	}
 
-	bool VirtualTerminalClient::send_command(const std::vector<std::uint8_t> &data) const
+	bool VirtualTerminalClient::send_command(const std::vector<std::uint8_t> &data)
 	{
+		if (commandAwaitingResponse && (!SystemTiming::time_expired_ms(lastCommandTimestamp_ms, 1500)))
+		{
+			return false;
+		}
```

Comment:
I'd like suggest maybe something like this?

```suggestion
		bool isTimedOut = SystemTiming::time_expired_ms(lastCommandTimestamp_ms, 1500);
		
		if (commandAwaitingResponse && (!isTimedOut ))
		{
			return false;
		}
		else if ((0 != lastCommandTimestamp_ms) && isTimedOut)
		{
			CANStackLogger::warn("[VT]: Server response to a command timed out");
		}
```

Reply:
Logging is indeed desired here, I changed up your suggestion a little bit. I feel like it's easier to read:
```c++
if (commandAwaitingResponse)
{
	if (SystemTiming::time_expired_ms(lastCommandTimestamp_ms, 1500))
	{
		CANStackLogger::warn("[VT]: Server response to a command timed out");
		commandAwaitingResponse = false;
	}
	else
	{
		// We're still waiting for a response to the last command, so we can't send another one yet
		return false;
	}
}
```